### PR TITLE
Refactor typescript

### PR DESCRIPTION
**Pull request by ahmetbozaci**
- Added snowpack for using the UUID library.
- Change all files using typescript and separate other files

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { v4 as uuidV4 } from 'uuid';
+import Store from "./store";
+
+insertTime();
+
+//! Event: Display Books
+document.addEventListener('DOMContentLoaded', UI.displayBooks);
+
+//! Event: Add a Book
+const form = document.querySelector('form') as HTMLFormElement
```
Comment (barisxco) [inline src/index.ts]:
```ts
const form: HTMLFormElement = document.querySelector('form') as HTMLFormElement
```

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+
+  const titleInput = document.getElementById('title') as HTMLInputElement;
```
Comment (barisxco) [inline src/index.ts]:
```ts
const form: HTMLFormElement = document.querySelector('form') as HTMLFormElement
```

File: src/library.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import type Book from "./types";
+
+export default class Library implements Book {
+  title;
+  author;
+  id;
```
Comment (barisxco) [inline src/library.ts]: you can give type annotation in here also

File: src/store.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+export default class Store {
+
+  static getBooks() {
+    const books: string | null = localStorage.getItem('books')
+    const bookArray = books ? JSON.parse(books) : [];
+    return bookArray;
+  }
+
+  static setItem(books: Book):void {
+    localStorage.setItem('books', JSON.stringify(books));
```
Comment (barisxco) [inline src/store.ts]: we recommend `"book:"` prefix

File: src/store.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  static removeBook(id: string):void {
+    const books = Store.getBooks();
+    books.forEach((book: Book, index: number) => {
+      if (book.id === id) {
+        books.splice(index, 1)
+      }
+    })
+    Store.setItem(books)
+  }
```
Comment (barisxco) [inline src/store.ts:29]: why you need book id as object? you can do it only with array, every index can be id for you

File: src/types.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+type Book = {
+  id: string;
+  title: string;
+  author: string;
+}
+
+export default Book;
```
Comment (barisxco) [inline src/types.ts:7]: all types can be here

Comment (ahmetbozaci) [reply src/types.ts:7]: @barisx  Thank you for all your feedback

**Pull request merged**